app/scripts/modules/core/src/presentation/icons/Icon.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { memo } from 'react';
+
+import { iconsByName } from './iconsByName';
+
+export type IconNames = keyof typeof iconsByName;
+
+export interface IIconProps {
+  name: IconNames;
+  color: string;
+  customColor?: string;
+  size?: 'small' | 'medium' | 'large';
+  customSize?: string;
```

Comment:
I think we could reduce the need for some of the props like `customColor` and `customSize`:

- `color`: I like the idea of this component determining if the color is custom or not. I am not sure how feasible that is. Maybe we can just have the user input hex or a common color string, since we are using inline styling. Maybe we lookup to see if that custom string exists. Or maybe we leave it :) Just a suggestion, that could make this simpler.
- `size`: If the `pxDimensionsBySize[size]` does not exist, fallback to input size or default size. Thus reducing the need for `customSize`:   
```js
const width = pxDimensionsBySize[size] || size || pxDimensionsBySize[DEFAULT_SIZE]
```

A third thought, that I liked a lot when working with CSS-in-JS and I think could apply here is a catch-all `customStyles` prop that can override anything.

```js
style={{ width, fill, ...customStyles }}
```

Reply:
I'm just gonna respond to these in order:
- `color`: I actually agree wholeheartedly, but couldn't really figure out a way to make it work without doing inlining of actual CSS colors as you mention. We don't currently have a great story around getting available color values in JS code to check against, though I'm still holding out hope for some kind of CSS-in-JS solution to that problem. While not violently opposed to using CSS colors in the API, I also do believe it's important to make configuring colors as prescriptive and straightforward as possible for things like plugins, or PRs from folks who don't know much of anything about deck or its conventions. What if... we just omitted `customColor` for now and tried to move everything we migrate over to a color that lives in `styleguide`? I don't actually think there's a lot of places we'd run into one-off colors.
- `size`: Also agree, and I think the reason I didn't go this route was because I had already split out `customColor` and thought eh, what's another 'custom' prop 😆. I'm gonna just switch to using your suggested behavior
- `style`: I suspect we'll end up adding this. It's a pretty common convention to have both `className` and `style` passthroughs on these sorts of components, and I already added `className` as I ran into some uses for it. My preference would be to hold off on actually adding it until we try to move a thing that we think needs it — IMO the more we can learn and experiment with the API before adding surface area, the better the quality of those decisions will be.